src/test/java/com/microsoft/jenkins/containeragents/builders/AciDeploymentTemplateBuilderTest.java

```diff
@@ -0,0 +1,74 @@
+package com.microsoft.jenkins.containeragents.builders;
+
+
+import com.microsoft.jenkins.containeragents.aci.AciAgent;
+import com.microsoft.jenkins.containeragents.aci.AciCloud;
+import com.microsoft.jenkins.containeragents.aci.AciContainerTemplate;
+import com.microsoft.jenkins.containeragents.aci.AciPrivateIpAddress;
+import hudson.model.Descriptor;
+import hudson.slaves.RetentionStrategy;
+import hudson.slaves.SlaveComputer;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static java.util.Collections.emptyList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class AciDeploymentTemplateBuilderTest {
+
+    static Jenkins jenkinsMock = mock(Jenkins.class);
+
+    AciAgent agentMock = mock(AciAgent.class);
+
+
+    @BeforeClass
+    public static void setupClass(){
+        MockedStatic<Jenkins> staticMock = Mockito.mockStatic(Jenkins.class);
```

Comment:
normally you use a try-with-resources with `MockedStatic` should this be mocked inside the class or released in an `AfterClass` ?

tbh it would be nice to refactor so you don't need these singletons but might be a bit more than you are after.

The [plugin-util-api-plugin](https://github.com/jenkinsci/plugin-util-api-plugin) provides some good facades to make testing easier, e.g. https://github.com/jenkinsci/plugin-util-api-plugin/blob/master/src/main/java/io/jenkins/plugins/util/JenkinsFacade.java

Reply:
I had to put in `setupClass` because otherwise only one test runs successfully. But I didn't know about the plugin-util-api-plugin. I will refactor it. Thank you for your advice.

Reply:
I think using a try with resources possibly combined with ensuring the parent Pom is on the latest version would’ve fixed that specific error but using the facade should end up with nicer code

Reply:
Using the latest version or the parent POM ends in breaking enforcer rules. 

```shell
[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for javax.annotation:javax.annotation-api:1.2 [provided] paths to dependency are:
+-org.jenkins-ci.plugins:azure-container-agents:9999-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-core:2.277.2 [provided]
    +-org.kohsuke.stapler:stapler:1.262.1 [provided] (managed) <-- org.kohsuke.stapler:stapler:1.262.1 [provided]
      +-javax.annotation:javax.annotation-api:1.2 [provided]
and
+-org.jenkins-ci.plugins:azure-container-agents:9999-SNAPSHOT
  +-org.jenkins-ci.plugins:windows-azure-storage:359.v09573e5ff73f
    +-org.jenkins-ci.plugins:copyartifact:1.46
      +-org.jenkins-ci.plugins:matrix-project:1.18 (managed) <-- org.jenkins-ci.plugins:matrix-project:1.14
        +-org.jenkins-ci.plugins:junit:1.51 (managed) <-- org.jenkins-ci.plugins:junit:1.29
          +-io.jenkins.plugins:plugin-util-api:2.3.0 (managed) <-- io.jenkins.plugins:plugin-util-api:1.7.1
            +-javax.annotation:javax.annotation-api:1.3.2 [provided]
and
+-org.jenkins-ci.plugins:azure-container-agents:9999-SNAPSHOT
  +-org.jenkins-ci.plugins:windows-azure-storage:359.v09573e5ff73f
    +-org.jenkins-ci.plugins:copyartifact:1.46
      +-org.jenkins-ci.plugins:matrix-project:1.18 (managed) <-- org.jenkins-ci.plugins:matrix-project:1.14
        +-org.jenkins-ci.plugins:junit:1.51 (managed) <-- org.jenkins-ci.plugins:junit:1.29
          +-io.jenkins.plugins:plugin-util-api:2.3.0 (managed) <-- io.jenkins.plugins:plugin-util-api:1.7.1
            +-edu.hm.hafner:codingstyle:2.6.0 [provided]
              +-javax.annotation:javax.annotation-api:1.3.2 [provided]

```

IMO dependency update should be done in a separate PR. 

Unfortunately, `JenkinsFacade` does not provide all methods that are used in `AciDeploymentTemplateBuilder`. Do you have a hint for alternatives to `Jenkins.get().getLegacyInstanceId()` and `Jenkins.get()`?

Reply:
I assume it's a matter of PRing it in but cc @uhafner

Reply:
I think this should be used instead based off of JavaDoc: `InstanceIdentity.get().getPublic().getEncoded()` may want to create a facade here or in plugin util api plugin

Reply:
Thanks @timja, I will start with a facade here and in a second step I would prepare a PR to `plugin-util-api-plugin`.